server/lib/routes/get-frontend-pairing.js

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+// This route handler prevents REFRESH behaviour for the pairing flow
+// If the user refreshes the browser during pairing, we instruct them to start over
+
+module.exports = function () {
+  // The array is converted into a RegExp
+  const PAIRING_ROUTES = [
+    'pair/auth/allow',
+    'pair/auth/complete',
+    'pair/auth/wait_for_supp',
+    'pair/supp/allow',
+    'pair/supp/wait_for_auth',
+  ].join('|'); // prepare for use in a RegExp
+
+  return {
+    method: 'get',
+    path: new RegExp('^/(' + PAIRING_ROUTES + ')/?$'),
+    process: function (req, res) {
+      res.redirect(302, '/pair/failure');
```

Comment:
Is this redirect logged automatically so that we can see how frequently this occurs?

Reply:
> I imagine not since those are the starting points for the pairing flow.

Correct!

>Is this redirect logged automatically so that we can see how frequently this occurs?

Yeap, something like `route GET /pair/supp/allow 0.362 302` will be in Stackdriver